rclcpp/test/test_qos.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "rclcpp/qos.hpp"
+
+/*
+  Test the equality operator for QoS profiles.
+ */
+TEST(TestQoS, equality_basic) {
```

Comment:
Consider splitting this into multiple test cases to avoid the side effect of re-using `profile_a` and `profile_b`.
Why line 32 is true w/o resetting 5 to 10 is non-obvious to me for instance.

Reply:
i'll consider it. for context `keep_all` sets `depth` to 0 because it is no longer meaningful when using `RMW_QOS_POLICY_HISTORY_KEEP_ALL`